proxy/handler.go

```diff
@@ -0,0 +1,52 @@
+package proxy
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/ONSdigital/dp-legacy-cache-proxy/config"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func (proxy *Proxy) Handler(ctx context.Context, w http.ResponseWriter, req *http.Request, cfg *config.Config) {
+	targetURL := cfg.BabbageURL + req.URL.String()
+
+	client := &http.Client{}
+
+	proxyReq, err := http.NewRequestWithContext(ctx, req.Method, targetURL, req.Body)
+	if err != nil {
+		log.Error(ctx, "proxyHandler: Error constructing proxy request", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
By passing `err.Error()`, doesn't that mean that the end user will be able to see that message? If that's the case, we need to change this everywhere, because we don't want to leak implementation details to the user. A generic "Internal Server Error" would be enough. The specific error is useful for ONS, and you're already logging it in the line above.